Add navigation and diagnostics docs to new format (#5)
wso2/docs-ballerina-visual-studio-code-plugin: en/docs/edit-the-code/diagnostics.md

# Diagnostics

> The diagnostics show you the syntax and semantic errors in the source code. Varieties of diagnostics such as warnings and errors will be shown. They will show up in multiple places:

Reviewer: ```suggestion
Diagnostics are the syntax and semantic errors that are identified in the source code. These may include a variety of errors and warnings. Diagnostics will show up in multiple places:
```

* In the Status Bar, there is a summary of all errors and warnings counts.

> * You can click on the summary or view the PROBLEMS panel with a list of all current errors.

Reviewer: ```suggestion
* The summary view and the PROBLEMS panel lists all of the current errors.
```

> * If you open a file that has errors or warnings, they will be rendered inline with the text and in the overview ruler.

Reviewer: ```suggestion
* Errors or warnings in a Ballerina file will be rendered inline with the text and in the overview ruler.
```

Reviewer: Curious how errors and warning are rendered inline. Are we referring to the light bulb?
Also, will see if we can avoid the term `Overview ruler`.

Author: Redered inline here means the `red/yellow color underline` you will see if there's an error in your code.
![Screenshot from 2023-01-09 14-22-48](https://user-images.githubusercontent.com/11285165/211270498-295cc392-732a-4118-8799-6e1c9c1622e9.png)

Reviewer: ack

![Diagnostics](/en/docs/img/edit-the-code/diagnostics/diagnostics.png?raw=true)

> ## Quickfixes

Reviewer: ```suggestion
## Quick fixes
```

> For a selected set of diagnostics, you can see the quick fixes. For example, the `variable assignment is required` diagnostic will have 2 associated quick fixes to create a new variable and ignore the return value.

Reviewer: ```suggestion
For certain diagnostics, the possible quick fixes are suggested as code actions. You can easily select and apply the required quick fix from the list. 

For example, the `variable assignment is required` diagnostic suggests two quick fixes as follows: 
- Create a new variable
- Ignore the return value
```
